### DISCUSSION TITLE
set onClick/set action on custom component

### DISCUSSION THREAD

**Discussion by DevrayePatil** [Budibase/budibase#8346] [Help] [answered]
I have created custom plugin in that plugin I want to set action through the budibase settings panel 
How can I get this 
![image](https://user-images.githubusercontent.com/54846160/197173149-dd6983fc-e2d9-4a1a-95f3-fb0898cc7e9b.png)

I want to set setting on my custom plugin as shown in above diagram

**Comment by melohagan** [comment, marked as answer]
Hey @DevrayePatil 

I've added an example to the docs: https://docs.budibase.com/docs/custom-component#example-event-handler

In your case, the schema.json would include the following setting:

```json
{
  "type": "event",
  "label": "On Click",
  "key": "onClick"
}
```

Then you would put the listener on one of your svelte elements, e.g.

```html
<script>
export let onClick
</script>

<div>
  <button on:click={onClick} />
</div>
```
